text/0000-stabilize-arbitrary-self-types.md

```diff
@@ -442,34 +468,6 @@ fn main() {
 A previous PR based on the `Deref` alternative has been proposed before https://github.com/rust-lang/rfcs/pull/2362
 and was postponed with the expectation that the lang team would [get back to `arbitrary_self_types` eventually](https://github.com/rust-lang/rfcs/pull/2362#issuecomment-527306157).
 
-# Unresolved questions
-[unresolved-questions]: #unresolved-questions
-
-- With the proposed design, it is not possible to be generic over the receiver while permitting the plain `Self` to be slotted in:
-    ```rs
-    use std::ops::Receiver;
-
-    struct Foo(u32);
-    impl Foo {
-        fn get<R: Receiver<Target=Self>>(self: R) -> u32 {
-            self.0
-        }
-    }
-
-    fn main() {
-        let mut foo = Foo(1);
-        foo.get::<&Foo>(); // Error
-    }
-    ```
-    This fails, because `T: Receiver<Target=T>` generally does not hold.
-    An alternative would be to lift the associated type into a generic type parameter of the `Receiver` trait, that would allow adding a blanket `impl Receiver<T> for T` without overlap.
-- This sinister TODO is present in the code:
-    ```
-                // FIXME(arbitrary_self_types): We probably should limit the
-                // situations where this can occur by adding additional restrictions
-                // to the feature, like the self type can't reference method substs.
-    ```
```

Comment:
Is this really the same as the first question?

Reply:
Surprisingly, yes. This TODO is triggered only in cases where the self type and the receiver type end up not matching, and the only case I can trigger is the case given in the first question.

There's a single Rust test which enters this TODO code, and it's the one whose code matches the first question.

Reply:
Nice!